esphome/components/toshiba/toshiba.cpp

```diff
@@ -126,14 +422,45 @@ void ToshibaClimate::setup() {
   this->minimum_temperature_ = this->temperature_min_();
   this->maximum_temperature_ = this->temperature_max_();
   this->swing_modes_ = this->toshiba_swing_modes_();
+
+  // Ensure swing mode is always initialized to a valid value
+  if (this->swing_modes_.empty()) {
+    // No swing support for this model
+    this->swing_mode = climate::CLIMATE_SWING_OFF;
+  } else if (this->swing_modes_.find(this->swing_mode) == this->swing_modes_.end()) {
+    // Current swing mode not supported, reset to OFF
+    this->swing_mode = climate::CLIMATE_SWING_OFF;
+  }
+
+  // Ensure mode is valid - ESPHome should only use standard climate modes
+  if (this->mode != climate::CLIMATE_MODE_OFF && this->mode != climate::CLIMATE_MODE_HEAT &&
+      this->mode != climate::CLIMATE_MODE_COOL && this->mode != climate::CLIMATE_MODE_HEAT_COOL &&
+      this->mode != climate::CLIMATE_MODE_DRY && this->mode != climate::CLIMATE_MODE_FAN_ONLY) {
+    ESP_LOGW(TAG, "Invalid mode detected during setup, resetting to OFF");
+    this->mode = climate::CLIMATE_MODE_OFF;
+  }
+
+  // Ensure fan mode is valid
+  if (!this->fan_mode.has_value()) {
+    ESP_LOGW(TAG, "Fan mode not set during setup, defaulting to AUTO");
+    this->fan_mode = climate::CLIMATE_FAN_AUTO;
+  }
+
   // Never send nan to HA
   if (std::isnan(this->target_temperature))
     this->target_temperature = 24;
+
+  // Log final state for debugging HA errors
+  ESP_LOGI(TAG, "Setup complete - Mode: %d, Fan: %s, Swing: %d, Temp: %.1f", static_cast<int>(this->mode),
+           this->fan_mode.has_value() ? std::to_string(static_cast<int>(this->fan_mode.value())).c_str() : "NONE",
+           static_cast<int>(this->swing_mode), this->target_temperature);
```

Comment:
> Log final state for debugging HA errors

Please do not add messages like this as "info" level. If this level of debugging is required, expect users will enable "verbose" level to troubleshoot. See our [logging guidelines](https://developers.esphome.io/architecture/logging/#understanding-logger-overhead).

```suggestion
  // Log final state for debugging HA errors
  ESP_LOGV(TAG, "Setup complete - Mode: %d, Fan: %s, Swing: %d, Temp: %.1f", static_cast<int>(this->mode),
           this->fan_mode.has_value() ? std::to_string(static_cast<int>(this->fan_mode.value())).c_str() : "NONE",
           static_cast<int>(this->swing_mode), this->target_temperature);
```